docker/Dockerfile-prod

```diff
@@ -1,21 +1,65 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
 FROM python:3.5-alpine
 
-RUN apk add --update gcc musl-dev libffi-dev python-dev openssl-dev
+MAINTAINER mars@mozilla.com
+# These are unlikely to change from version to version of the container
+EXPOSE 9000
+CMD ["/usr/local/bin/uwsgi"]
 
-RUN addgroup -g 1001 app && \
-    adduser -D -u 1001 -G app -s /usr/sbin/nologin app
+RUN addgroup -g 10001 app && adduser -D -u 10001 -G app -h /app app
 
-WORKDIR /app
 COPY requirements.txt /requirements.txt
-RUN pip install -r /requirements.txt
 
-RUN apk del gcc musl-dev libffi-dev python-dev openssl-dev
+# Install pure-Python, compiled, and OS package dependencies.  Use scanelf to
+# uninstall any compile-time OS package dependencies and keep only the run-time
+# OS package dependencies.
+RUN set -ex \
+    && apk add --update --no-cache --virtual .build-deps \
+           gcc \
+           libc-dev \
+           musl-dev \
+           linux-headers \
+           pcre-dev \
+           openssl-dev \
+           libffi-dev \
+    && pip install --no-cache -r /requirements.txt \
+    && pip install --no-cache uWSGI==2.0.15 \
```

Comment:
Come to think of it, we have so many requirements files now, we might want to switch to a cleaner directory structure: add a `requirements/` directory, and put `dev.txt`, `common.txt`, `circle.txt`, and `production.txt` requirements files in there.